core/src/main/java/io/grpc/transport/Http2ClientStream.java

```diff
@@ -136,8 +136,7 @@ protected void transportDataReceived(Buffer frame, boolean endOfStream) {
       if (endOfStream) {
         // This is a protocol violation as we expect to receive trailers.
         transportError = Status.INTERNAL.withDescription("Recevied EOS on DATA frame");
-        frame.close();
-        inboundTransportError(transportError);
+        inboundTransportError(transportError, true);
```

Comment:
Why did this need to stop delivery? Shouldn't we just have removed the frame.close() line and then the deframer would be closed (closing its frames) in AbstractClientStream.closeListener()?